dspy/teleprompt/random_search.py

```diff
@@ -100,6 +100,10 @@ def compile(self, student, *, teacher=None, trainset, valset=None):
 
             print(f"Scores so far: {scores}")
 
+            if self.stop_at_score is not None and score >= self.stop_at_score:
+                print(f"Stopping early because score {score} is >= stop_at_score {self.stop_at_score}")
+                break
+
```

Comment:
Shouldn’t this new block be moved to the very end of the loop code. In particular: after updating the best program?